src/main/java/io/numaproj/numaflow/info/ServerInfoAccessorImpl.java

```diff
@@ -0,0 +1,51 @@
+package io.numaproj.numaflow.info;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+@AllArgsConstructor
+public class ServerInfoAccessorImpl implements ServerInfoAccessor {
+    private ObjectMapper objectMapper;
+
+    @Override
+    public String getSDKVersion() {
+        // This only works for Java 9 and above.
+        // Since we already use 11+ for numaflow SDK, it's safe to apply this approach.
+        return String.valueOf(Runtime.version().version().get(0));
+    }
+
+    @Override
+    public void write(ServerInfo serverInfo, String filePath) throws IOException {
+        File file = new File(filePath);
+        if (file.exists()) {
+            file.delete();
+        }
+        FileWriter fileWriter = new FileWriter(filePath, false);
+        objectMapper.writeValue(fileWriter, serverInfo);
+        FileWriter eofWriter = new FileWriter(filePath, true);
+        eofWriter.append("\n").append(ServerInfoConstants.EOF);
+        eofWriter.close();
+        fileWriter.close();
+    }
+
+    @Override
+    public ServerInfo read(String filePath) throws IOException {
+        File file = new File(filePath);
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+        StringBuilder stringBuilder = new StringBuilder();
+        String line;
+        while ((line = bufferedReader.readLine()) != null
+                && !line.equals(ServerInfoConstants.EOF)) {
```

Comment:
I guess `\n` is for this.

I think it's better to use `endWith()` to check and trim.

Reply:
We don't have to add `\n`, I just feel it looks prettier when EOF is at a newline :). I also made corresponding changes in https://github.com/numaproj/numaflow-go/pull/48 to make sure the parsing is consistent.